library/std/src/sys/unix/fd.rs

```diff
@@ -26,6 +27,35 @@ const READ_LIMIT: usize = c_int::MAX as usize - 1;
 #[cfg(not(target_os = "macos"))]
 const READ_LIMIT: usize = libc::ssize_t::MAX as usize;
 
+#[cfg(any(target_os = "linux", target_os = "macos"))]
```

Comment:
This shouldn't be restricted to just Linux and macOS, all Unix-like OSes have limits on iovec length. Also it seems that all these targets have a definition of `_SC_IOV_MAX` so there is no need to special case `_SC_UIO_MAXIOV` for macOS.

Reply:
> This shouldn't be restricted to just Linux and macOS, all Unix-like OSes have limits on iovec length.

What I was wondering here is whether `#[cfg(unix)]` implies POSIX and hence allows relying on `sysconf` and friends? Will change the code under that assumption...

> so there is no need to special case _SC_UIO_MAXIOV for macOS.

I am admittedly just not knowledgeable on macOS and took that information from the linked bug report. Will change both to just use `_SC_IOV_MAX`.